core/src/main/java/io/smallrye/concurrency/SmallRyeConcurrencyManager.java

```diff
@@ -142,8 +150,19 @@ ThreadContextProviderPlan getProviderPlan(String[] propagated, String[] unchange
 		return new ThreadContextBuilderImpl(this);
 	}
 
+	public Set<String> getAllThreadContextProviderNames() {
+		Set<String> result = providers.stream().map(tcp -> tcp.getThreadContextType()).collect(Collectors.toSet());
+		// FIXME following contexts are to be default part of implementation and should always be known
+		// remove once implemented
+		result.add(ThreadContext.CDI);
+		result.add(ThreadContext.SECURITY);
+		result.add(ThreadContext.TRANSACTION);
+		result.add(ThreadContext.APPLICATION);
+		return result;
+	}
+
```

Comment:
What's the use for that method? It doesn't seem to be used. I've removed it temporarily.

Reply:
Validation needs to be done when calling `Builder.build()`.
I needed a method that will hand me over a Collection of all provider names.
It should be used in `ThreadContext` class, `verifyContextsExist()` method.

Potentially could be replaced by the method you had (which I only found later) that returns an array of names.
I just didn't get to it yet but was planning on merging those two methods and preferably keeping the one with set as opposed to arrays.

Reply:
OK I see where it was used and did it slightly differently. I think that if at some point we have to provide temporary/fake thread contexts, we should write a temporary no-op thread context plugin for them.